cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/monitor/LoadMonitorState.java

```diff
@@ -172,39 +172,41 @@ public static LoadMonitorState loading(int numValidSnapshotWindows,
 
   private void setCommonJsonGenericAttributeCollection(boolean verbose, Map<String, Object> loadMonitorState) {
     loadMonitorState.put(STATE, _loadMonitorTaskRunnerState);
-    loadMonitorState.put(TRAINED, ModelParameters.trainingCompleted());
-    loadMonitorState.put(TRAINING_PCT, ModelParameters.trainingCompleted()
-                                       ? 100.0 : (ModelParameters.modelCoefficientTrainingCompleteness() * 100));
-    loadMonitorState.put(NUM_MONITORED_WINDOWS, _monitoredWindows.size());
-    if (verbose) {
-      loadMonitorState.put(MONITORED_WINDOWS, _monitoredWindows);
+    if (_loadMonitorTaskRunnerState != LoadMonitorTaskRunnerState.NOT_STARTED) {
+      loadMonitorState.put(TRAINED, ModelParameters.trainingCompleted());
+      loadMonitorState.put(TRAINING_PCT, ModelParameters.trainingCompleted()
+                                         ? 100.0 : (ModelParameters.modelCoefficientTrainingCompleteness() * 100));
+      loadMonitorState.put(NUM_MONITORED_WINDOWS, _monitoredWindows.size());
+      loadMonitorState.put(MONITORED_WINDOWS, verbose ? _monitoredWindows : null);
+      loadMonitorState.put(NUM_VALID_PARTITIONS, _numValidMonitoredPartitions);
+      loadMonitorState.put(NUM_TOTAL_PARTITIONS, _totalNumPartitions);
+      loadMonitorState.put(MONITORING_COVERAGE_PCT, nanToZero(validPartitionsRatio() * 100));
+      loadMonitorState.put(NUM_FLAWED_PARTITIONS, _sampleExtrapolations.size());
+    } else {
+      loadMonitorState.put(TRAINED, null);
+      loadMonitorState.put(TRAINING_PCT, null);
+      loadMonitorState.put(NUM_MONITORED_WINDOWS, null);
+      loadMonitorState.put(MONITORED_WINDOWS, null);
+      loadMonitorState.put(NUM_VALID_PARTITIONS, null);
+      loadMonitorState.put(NUM_TOTAL_PARTITIONS, null);
+      loadMonitorState.put(MONITORING_COVERAGE_PCT, null);
+      loadMonitorState.put(NUM_FLAWED_PARTITIONS, null);
```

Comment:
This `nullifying` pattern seems to involve so much repetition and a little error prone. I am curious whether it is possible to generalize this by creating some common wrapper for response keyset and then building util functions to nullify them?

Reply:
I did thought about that, my concern there is many case we ends in setting the value for the same field twice(first `null` then real value).  The good side is that the code will be clearer. 
Looks like it is worthy the cost.:)